### PR TITLE
erasure-coding: do not panic on 1 validator

### PR DESCRIPTION
**Pull request by ordian**
as found in #2021, I don't think that was intended, please correct me if I'm wrong. We should return an error rather than panic.

### CODE REVIEW

File: erasure-coding/fuzzer/src/reconstruct.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use polkadot_erasure_coding::*;
+use primitives::v1::AvailableData;
+use honggfuzz::fuzz;
+
+fn main(){
```
Comment (pepyakin) [inline erasure-coding/fuzzer/src/reconstruct.rs:5]:
```suggestion
fn main() {
```

File: erasure-coding/fuzzer/src/round_trip.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+use polkadot_erasure_coding::*;
+use primitives::v1::{AvailableData, BlockData, PoV};
+use std::sync::Arc;
+use honggfuzz::fuzz;
+
+
+fn main(){
```
Comment (pepyakin) [inline erasure-coding/fuzzer/src/round_trip.rs:7]:
```suggestion
fn main() {
```

**Review by pepyakin — APPROVED** [review]
Not sure either, but looks good.

**Comment by rphmeier** [comment]
An alternative approach would be to alter `make_shards_for`'s behavior when `data shards == 1 && parity_shards == 0` to just return the same data. And to alter the recovery routine to do roughly the same. For now I'm OK merging this and we can revisit 1-node testnets later

**Pull request merged**